src/Sets/HParallelotope.jl

```diff
@@ -0,0 +1,311 @@
+    a combination of zonotopes and polytopes.* Nonlinear analysis: hybrid systems 4.2 (2010): 233-249.
+"""
+struct HParallelotope{N<:Real, VN<:AbstractVector{N}, MN<:AbstractMatrix{N}} <: AbstractZonotope{N}
+    directions::MN
+    offset::VN
+
+    # default constructor with dimension check
+    function HParallelotope(D::MN, c::VN) where {N<:Real, VN<:AbstractVector{N}, MN<:AbstractMatrix{N}}
+        @assert length(c) == 2*checksquare(D) "the length of the offset direction should be twice the size " *
+            "of the directions matrix, but they are $(length(c)) and $(size(D)) dimensional respectively"
+        return new{N, VN, MN}(D, c)
+    end
+end
+
+# =================
+# Getter functions
+# =================
+
+"""
+    directions(P::HParallelotope{N}) where {N<:Real}
+
+Return the directions matrix of a parallelotope.
+
+### Input
+
+- `P` -- parallelotope in constraint representation
+
+### Output
+
+A matrix where each row represents a direction of the parallelotope.
+The negated directions `-D_i` are implicit (see [`HParallelotope`](@ref) for details).
+"""
+function directions(P::HParallelotope{N}) where {N<:Real}
+    return P.directions
+end
+
+"""
+    offset(P::HParallelotope{N}) where {N<:Real}
+
+Return the offsets of a parallelotope.
+
+### Input
+
+- `P` -- parallelotope in constraint representation
+
+### Output
+
+A vector with the ``2n`` offsets of the parallelotope,
+if ``n`` is the dimension of ``P``.
+"""
+function offset(P::HParallelotope{N}) where {N<:Real}
+    return P.offset
+end
+
+# ========================================
+# Functions required by the set interface
+# ========================================
+
+"""
+    dim(P::HParallelotope{N}) where {N<:Real}
+
+Return the dimension of a parallelotope.
+
+### Input
+
+- `P` -- parallelotope in constraint representation
+
+### Output
+
+An integer representing the dimension of the parallelotope
+"""
+function dim(P::HParallelotope{N}) where {N<:Real}
+    return size(P.directions, 1)
+end
+
+# ========================
+# Type-specific functions
+# ========================
+
+"""
+    base_vertex(P::HParallelotope{N}) where {N<:Real}
+
+Compute the base vertex of the given parallelotope in constraint representation.
+
+### Input
+
+- `P` -- parallelotope in constraint representation
+
+### Output
+
+The base vertex of ``P``.
+
+### Algorithm
+
+Intuitively, the base vertex is the point on which the generators are anchored.
+The base vertex can be computed as the solution of the ``n``-dimensional linear
+system ``D_i x = c_{n+i}`` for ``i = 1, \\ldots, n``, see [1, Section 3.2.1].
+
+[1] Dreossi, Tommaso, Thao Dang, and Carla Piazza. *Reachability computation for
+    polynomial dynamical systems.* Formal Methods in System Design 50.1 (2017): 1-38.
+"""
+function base_vertex(P::HParallelotope{N}) where {N<:Real}
+    D, c = P.directions, P.offset
+    n = dim(P)
+    v = to_negative_vector(c[n+1:end]) # converts to a dense vector as well
```

Comment:
can we use a view here? `view(c, (n+1):length(c))`

Reply:
Yes, it improved the performance a little bit :)